### PR TITLE
Fixing the gitlab master ci

### PR DESCRIPTION
**Pull request by jamiebramwell**
This adds a mirror option to the gitlab master CI in case TPL repos are inaccessible.

### CODE REVIEW

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
LGTM

File: .gitlab/ci/build_quartz.yml [hunk 1]
```diff
@@ -67,16 +67,19 @@ clang_9_0_0 (Master build_with_deps_on_quartz):
   variables:
     COMPILER: "clang@9.0.0"
     SPEC: "@develop%${COMPILER}"
+    MIRROR: "/usr/workspace/smithdev/mirror"
```
Comment (adrienbernede) [inline .gitlab/ci/build_quartz.yml:70]: This variable should be declared globally, avoiding repetition.
See 'variables' section in  `.gitlab-ci.yml`.

Comment (jamiebramwell) [reply .gitlab/ci/build_quartz.yml]: Thanks for the tip! I'm slowly learning how all of this fits together...

**Comment by jamiebramwell** [comment]
LGTM

File: .gitlab/ci/build_quartz.yml (hunk 1 above)
Comment (adrienbernede) [reply .gitlab/ci/build_quartz.yml]: That’s what reviews are for :)

**Comment by jamiebramwell** [comment]
> Maybe you can test it and post the link to the successful pipeline.
> 
> Sorry if I am slowing you down, but I think it’s good thing to know.

Oh no worries! I actually tested exactly that on the smith repo, but then realized we could just put it here instead. It's probably good to check again anyway! The pipeline is going now!

**Comment by adrienbernede** [comment]
Cool! I didn’t know if you were aware of the mechanism.
You have my approval anyway ;)

**Pull request merged**